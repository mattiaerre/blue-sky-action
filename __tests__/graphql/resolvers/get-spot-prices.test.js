const getSpotPrices = require('../../../src/graphql/resolvers/get-spot-prices');
// const currency = require('../../../src/data/currency');

test('to match snapshot', async () => {
  const args = { currency: 'USD' };
  const context = {
    config: { coinbase: { baseUrl: 'https://api.coinbase.com' } }
  };

  const spotPrices = (
    await getSpotPrices(null, args, context)
  ).map(({ amount, ...rest }) => ({ ...rest }));

  expect(spotPrices).toMatchSnapshot();

  // console.log(spotPrices.filter(price => currency[price.base] === undefined));
});
