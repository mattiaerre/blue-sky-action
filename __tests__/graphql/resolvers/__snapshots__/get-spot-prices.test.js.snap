// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snapshot 1`] = `
Array [
  Object {
    "base": "BTC",
    "currency": "USD",
  },
  Object {
    "base": "BCH",
    "currency": "USD",
  },
  Object {
    "base": "BSV",
    "currency": "USD",
  },
  Object {
    "base": "ETH",
    "currency": "USD",
  },
  Object {
    "base": "ETC",
    "currency": "USD",
  },
  Object {
    "base": "LTC",
    "currency": "USD",
  },
  Object {
    "base": "ZRX",
    "currency": "USD",
  },
  Object {
    "base": "USDC",
    "currency": "USD",
  },
  Object {
    "base": "BAT",
    "currency": "USD",
  },
  Object {
    "base": "KNC",
    "currency": "USD",
  },
  Object {
    "base": "LINK",
    "currency": "USD",
  },
  Object {
    "base": "DAI",
    "currency": "USD",
  },
  Object {
    "base": "SAI",
    "currency": "USD",
  },
  Object {
    "base": "ZEC",
    "currency": "USD",
  },
  Object {
    "base": "XRP",
    "currency": "USD",
  },
  Object {
    "base": "REP",
    "currency": "USD",
  },
  Object {
    "base": "XLM",
    "currency": "USD",
  },
  Object {
    "base": "EOS",
    "currency": "USD",
  },
  Object {
    "base": "XTZ",
    "currency": "USD",
  },
  Object {
    "base": "DASH",
    "currency": "USD",
  },
  Object {
    "base": "ATOM",
    "currency": "USD",
  },
  Object {
    "base": "OXT",
    "currency": "USD",
  },
  Object {
    "base": "EUR",
    "currency": "USD",
  },
  Object {
    "base": "GBP",
    "currency": "USD",
  },
  Object {
    "base": "CAD",
    "currency": "USD",
  },
  Object {
    "base": "JPY",
    "currency": "USD",
  },
]
`;
